### ISSUE TITLE
Plugin for multi-release JAR building and testing

### ISSUE THREAD

**Issue by chaosflaws** [gradlex-org/java-module-dependencies#80]
I have provided a [reproducible sample](https://github.com/chaosflaws/mrjars-test).

I tried to test a multi-release jar (17 and 21) for behavioral consistency by applying the existing unit tests first to the java 17 sources and then to the combined java 17 + java 21 sources.

The new test suite is called `test21`, which in turn creates a source set of that name. Because of this, `java-module-dependencies` complains that the module name - in this case, `de.chaosflaws.mrjars.test` - does not match the new source set name. (But it's the same _sources_ - once as `test` and once as `test21`.)

This could be remedied if test suites could reuse existing source sets. (I opened an [issue](https://github.com/gradle/gradle/issues/27758).) However, I don't expect gradle to address this soon.

Is there something that makes sense to implement here to avoid the problem I am facing?

**Comment by chaosflaws** [comment]
Sorry for taking so long to respond. I did manage to get the MR-JAR setup to work using your solution. I had to use a bit of exception handling around the toolchains approach, since not all Java versions are available on all my CI executors, but other than that, it was (now) pretty straightforward. Thanks!

> I think it would make sense to provide a (small) plugin that makes the setup easier. Probably as a separate GradleX (?) plugin. I'll give it some thought.

As someone who has become quite familiar with Gradle, but is by no means an expert: Your code was certainly helpful to me. At the same time, the topic of Multi-Release Jars does not seem to have caught on much since the feature became available in Java 9. This may be because Gradle does not offer great support for it, but admittedly, other parts of the Java ecosystem do not really provide support for it either. It's hard to say if there's no need or if it's just too hard to set up for the average developer.

The original issue can be considered "resolved".
